targets/zephyr/src/zephyr_getline.c

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2016 Linaro
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <zephyr.h>
+#include <uart.h>
+#include <drivers/console/uart_console.h>
+#include "zephyr_getline.h"
+
+/* While app processes one input line, Zephyr will have another line
+   buffer to accumulate more console input. */
+static struct uart_console_input line_bufs[2];
+
+static struct nano_fifo free_queue;
+static struct nano_fifo used_queue;
+
+char *zephyr_getline(void)
+{
+	static struct uart_console_input *cmd;
+
+	/* Recycle cmd buffer returned previous time */
+	if (cmd != NULL) {
```

Comment:
(sytle) General comment to this whole file: it does not follow project style guidelines (2 space indentation, curly braces on separate lines, space before parentheses, etc.) Being a port implementation, I'm tempted not to be too picky, but the other files in this port do follow the style, so I guess this one should do too.